homeassistant/components/nzbget/__init__.py

```diff
@@ -52,147 +55,213 @@
                 vol.Optional(
                     CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
                 ): cv.time_period,
-                vol.Optional(CONF_SSL, default=False): cv.boolean,
+                vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
             }
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
+SPEED_LIMIT_SCHEMA = vol.Schema(
+    {vol.Optional(ATTR_SPEED, default=DEFAULT_SPEED_LIMIT): cv.positive_int}
+)
+
 
-def setup(hass, config):
-    """Set up the NZBGet sensors."""
+async def async_setup(hass: HomeAssistantType, config: dict) -> bool:
+    """Set up the NZBGet integration."""
+    hass.data.setdefault(DOMAIN, {})
 
-    host = config[DOMAIN][CONF_HOST]
-    port = config[DOMAIN][CONF_PORT]
-    ssl = "s" if config[DOMAIN][CONF_SSL] else ""
-    name = config[DOMAIN][CONF_NAME]
-    username = config[DOMAIN].get(CONF_USERNAME)
-    password = config[DOMAIN].get(CONF_PASSWORD)
-    scan_interval = config[DOMAIN][CONF_SCAN_INTERVAL]
+    if len(hass.config_entries.async_entries(DOMAIN)) > 0:
```

Comment:
```suggestion
    if hass.config_entries.async_entries(DOMAIN):
```

I don't think you need len() here